test.js

```diff
@@ -133,3 +133,12 @@ test('PCancelable.fn()', async t => {
 test('PCancelable.CancelError', t => {
 	t.true(PCancelable.CancelError.prototype instanceof Error);
 });
+
+test.failing('resolves on cancel', async t => {
+	const p = new PCancelable((onCancel, resolve, reject) => {
+		onCancel(() => {});
+	});
+	setTimeout(() => { p.cancel(); }, 100);
+	await t.throws(Promise.resolve(p));
```

Comment:
This won't work as `Promise.resolve()` creates a new promise from the existing one, so the `.cancel()` method is not preserved.